src/drivers/cholqrcp.cc

```diff
@@ -0,0 +1,249 @@
+    //-------TIMING VARS--------/
+    high_resolution_clock::time_point saso_t_stop;
+    high_resolution_clock::time_point saso_t_start;
+    long saso_t_dur = 0;
+
+    high_resolution_clock::time_point qrcp_t_start;
+    high_resolution_clock::time_point qrcp_t_stop;
+    long qrcp_t_dur = 0;
+    
+    high_resolution_clock::time_point rank_reveal_t_start;
+    high_resolution_clock::time_point rank_reveal_t_stop;
+    long rank_reveal_t_dur = 0;
+
+    high_resolution_clock::time_point cholqrcp_t_start;
+    high_resolution_clock::time_point cholqrcp_t_stop;
+    long cholqrcp_t_dur = 0;
+
+    high_resolution_clock::time_point a_mod_piv_t_start;
+    high_resolution_clock::time_point a_mod_piv_t_stop;
+    long a_mod_piv_t_dur = 0;
+
+    high_resolution_clock::time_point a_mod_trsm_t_start;
+    high_resolution_clock::time_point a_mod_trsm_t_stop;
+    long a_mod_trsm_t_dur = 0;
+
+    high_resolution_clock::time_point copy_t_start;
+    high_resolution_clock::time_point copy_t_stop;
+    long copy_t_dur = 0;
+
+    high_resolution_clock::time_point resize_t_start;
+    high_resolution_clock::time_point resize_t_stop;
+    long resize_t_dur = 0;
+
+    high_resolution_clock::time_point total_t_start;
+    high_resolution_clock::time_point total_t_stop;
+    long total_t_dur = 0;
+    //-------TIMING VARS--------/
+
+    //-------TIMING--------/
+    if(this -> timing) {
+        total_t_start = high_resolution_clock::now();
+    
+        resize_t_start = high_resolution_clock::now();
+    }
+    //-------TIMING--------/
+
+    T* A_dat       = A.data();
+    T* A_hat_dat   = upsize(d * n, this->A_hat);
+    T* tau_dat     = upsize(n, this->tau);
+    J.resize(n);
+    int64_t* J_dat = J.data();
+
+    //-------TIMING--------/
+    if(this -> timing) {
+        resize_t_stop = high_resolution_clock::now();
+        resize_t_dur = duration_cast<microseconds>(resize_t_stop - resize_t_start).count();
+    }
+    
+    //-------TIMING--------/
+    if(this -> timing) {
+        saso_t_start = high_resolution_clock::now();
+    }
+    //-------TIMING--------/
+
+    /*
+    Old way of generating and applying SASO
+
+    struct RandBLAS::sasos::SASO sas;
+    sas.n_rows = d; // > n
+    sas.n_cols = m;
+    sas.vec_nnz = this->nnz;
+    sas.rows = new int64_t[sas.vec_nnz * m];
+    sas.cols = new int64_t[sas.vec_nnz * m];
+    sas.vals = new double[sas.vec_nnz * m];
+    RandBLAS::sasos::fill_colwise(sas, this->seed, 0);
+
+    RandBLAS::sasos::sketch_csccol(sas, n, (double*) A_dat, (double*) A_hat_dat, this->num_threads);
+    */
+
+    RandBLAS::sparse::SparseDist DS = {RandBLAS::sparse::SparseDistName::SASO, d, m, this->nnz};        
+    RandBLAS::sparse::SparseSkOp<T> S(DS, this->seed, 0, NULL, NULL, NULL);        
+    RandBLAS::sparse::fill_sparse(S);
+
+    RandBLAS::sparse::lskges<T, RandBLAS::sparse::SparseSkOp<T>>(blas::Layout::ColMajor, blas::Op::NoTrans, blas::Op::NoTrans, d, n, m, 1.0, S, 0, 0, A.data(), m, 0.0, A_hat_dat, d);
```

Comment:
I'd appreciate if you broke this function call into a few lines, a-la,
```
RandBLAS::sparse::lskges<T, RandBLAS::sparse::SparseSkOp<T>>(
    blas::Layout::ColMajor, blas::Op::NoTrans, blas::Op::NoTrans,
    d, n, m, 1.0, S, 0, 0, A.data(), m, 0.0, A_hat_dat, d
);
```

Reply:
Separately, I'll note that you're already using the ``blas`` namespace, so the ``blas`` qualifier on ``Layout`` isn't strictly necessary. Feel free to just _not_ use the blas namespace though, and always use the ``blas::`` prefix instead.

Reply:
Fixed.